PhotoDetailScreen.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect } from 'react';
+import { StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
+import { RouteProp, useRoute, useNavigation } from '@react-navigation/native';
+
+type RootStackParamList = {
+  PhotoDetail: { imageUrl: string };
+  ImageModal: { imageUrl: string };
+};
+
+type PhotoDetailRouteProp = RouteProp<RootStackParamList, 'PhotoDetail'>;
+
+const PhotoDetailScreen = () => {
+  const route = useRoute<PhotoDetailRouteProp>();
+  const imageUrl = route.params.imageUrl;
+  const navigation = useNavigation();
+
+  useEffect(() => {
+    // Set the header title to the URL of the photo
+    navigation.setOptions({ title: imageUrl });
+  }, []); // Empty dependency array ensures this effect runs only once
+
+  const handleImageModalPress = () => {
+    console.log('Navigating to ImageModal:', imageUrl);
+    navigation.navigate('ImageModal', { imageUrl } as RootStackParamList['ImageModal']); // Navigator doesn't navigate unless this is in there, not sure why
```

Comment:
casting shouldn't have any impact on functionality. Typescript gives you an error because there wasn't a type applied to `useNavigation`